src/coupling.jl

```diff
@@ -0,0 +1,68 @@
+SplitCoupledJumpProblem(prob::AbstractJumpProblem,prob_control::AbstractJumpProblem,aggregator::AbstractAggregatorAlgorithm,coupling_map::Vector{Tuple{Int64,Int64}})= JumpProblem(cat_problems(prob,prob_control),aggregator,build_split_jumps(prob,prob_control,coupling_map)...)
+
+# make new discrete problem by joining initial_data
+function cat_problems(prob::AbstractJumpProblem,prob_control::AbstractJumpProblem)
+  coupled_u0 = cat(1,prob.prob.u0,prob_control.prob.u0)
+  DiscreteProblem(coupled_u0,prob.prob.tspan)
```

Comment:
> Does this need to be restrained to DiscreteProblem?

I don't think so, but I haven't experimented with that yet. 

> Is there any think that makes it so that way constant rate jumps in an ODE/SDE would not have the correct accuracy? I don't know this well enough.

If I understand what you are asking, then no. The coupling shouldn't really have a significant impact on the accuracy since the jump rates of the coupled process are comparable to the original process.